include/reactor-cpp/port.hh

```diff
@@ -18,122 +19,112 @@ namespace reactor {
+  [[nodiscard]] auto has_dependencies() const noexcept -> bool { return !dependencies_.empty(); }
+  [[nodiscard]] auto has_antidependencies() const noexcept -> bool { return !antidependencies_.empty(); }
 
-  BasePort* inward_binding() const { return _inward_binding; }
-  const auto& outward_bindings() const { return _outward_bindings; }
+  [[nodiscard]] auto inward_binding() const noexcept -> BasePort* { return inward_binding_; }
+  [[nodiscard]] auto outward_bindings() const noexcept -> const auto& { return outward_bindings_; }
 
-  const auto& triggers() const { return _triggers; }
-  const auto& dependencies() const { return _dependencies; }
-  const auto& antidependencies() const { return _antidependencies; }
+  [[nodiscard]] auto triggers() const noexcept -> const auto& { return triggers_; }
+  [[nodiscard]] auto dependencies() const noexcept -> const auto& { return dependencies_; }
+  [[nodiscard]] auto antidependencies() const noexcept -> const auto& { return antidependencies_; }
 
   friend class Reaction;
   friend class Scheduler;
 };
 
-template <class T>
-class Port : public BasePort {
- private:
-  ImmutableValuePtr<T> value_ptr{nullptr};
+template <class T> class Port : public BasePort {
+private:
+  ImmutableValuePtr<T> value_ptr_{nullptr};
 
-  void cleanup() override final { value_ptr = nullptr; }
+  void cleanup() final { value_ptr_ = nullptr; }
 
- public:
+public:
   using value_type = T;
 
   Port(const std::string& name, PortType type, Reactor* container)
       : BasePort(name, type, container) {}
 
   void bind_to(Port<T>* port) { base_bind_to(port); }
-  Port<T>* typed_inward_binding() const;
-  const std::set<Port<T>*>& typed_outward_bindings() const;
+  auto typed_inward_binding() const noexcept -> Port<T>*;
+  auto typed_outward_bindings() const noexcept -> const std::set<Port<T>*>&;
 
   void set(const ImmutableValuePtr<T>& value_ptr);
-  void set(MutableValuePtr<T>&& value_ptr) {
-    set(ImmutableValuePtr<T>(std::forward<MutableValuePtr<T>>(value_ptr)));
-  }
+  void set(MutableValuePtr<T>&& value_ptr) { set(ImmutableValuePtr<T>(std::forward<MutableValuePtr<T>>(value_ptr))); }
   void set(const T& value) { set(make_immutable_value<T>(value)); }
   void set(T&& value) { set(make_immutable_value<T>(std::forward<T>(value))); }
   // Setting a port to nullptr is not permitted.
   void set(std::nullptr_t) = delete;
 
-  void startup() override final {}
-  void shutdown() override final {}
-
-  const ImmutableValuePtr<T>& get() const;
-  bool is_present() const;
+  void startup() final {}
+  void shutdown() final {}
 
+  auto get() const noexcept -> const ImmutableValuePtr<T>&;
+  [[nodiscard]] auto is_present() const noexcept -> bool;
 };
 
-template <>
-class Port<void> : public BasePort {
- private:
-  bool present{false};
-
-  void cleanup() override final { present = false; }
+template <> class Port<void> : public BasePort {
+private:
+  bool present_ = false;
```

Comment:
Why did this change from brace initialization to equal initialization? As far as I understand, there are good reasons to use brace initialization wherever possible. See [this discussion](https://stackoverflow.com/questions/18222926/why-is-list-initialization-using-curly-braces-better-than-the-alternatives) for instance.